core/src/main/java/io/kestra/core/tasks/flows/Flow.java

```diff
@@ -119,6 +128,7 @@ public String flowUidWithoutRevision() {
     @SuppressWarnings("unchecked")
     public Execution createExecution(RunContext runContext, FlowExecutorInterface flowExecutorInterface) throws Exception {
         RunnerUtils runnerUtils = runContext.getApplicationContext().getBean(RunnerUtils.class);
+        ExecutionRepositoryInterface executionRepository = runContext.getApplicationContext().getBean(ExecutionRepositoryInterface.class);
```

Comment:
Please don't use the ExecutionRepository, you can instead change the method signature to include the parent execution and retrieve the information you want.
The only place where this method is called is [here](https://github.com/kestra-io/kestra/blob/develop/core/src/main/java/io/kestra/core/runners/ExecutorService.java#L582) so you can just add it there :+1: